src/overlays/actors/ovl_Bg_Jya_Lift/z_bg_jya_lift.c

```diff
@@ -26,24 +34,132 @@ const ActorInit Bg_Jya_Lift_InitVars = {
+    extern s32      D_8089A044;
+    extern s32      D_0600D7E8;
+    extern UNK_TYPE D_0600CCE0;
+
+void BgJyaLiftInitHelper(BgJyaLift* this, GlobalContext* globalCtx, u32 arg2, DynaPolyMoveFlag moveFlag) { //seems to help set up the actor
+    s32 pad1; // padding that makes everything match nicely
+    u32 localConst;
+    localConst = 0U;
+    DynaPolyInfo_SetActorMove(&this->dyna, moveFlag);
+    DynaPolyInfo_Alloc(arg2, (void*)&localConst);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna, localConst);
+}
+
+void BgJyaLift_Init(Actor* thisx, GlobalContext* globalCtx) {
+     BgJyaLift* this = THIS;
+    THIS->unk16A = 0;
+    if ((D_8089A020) != 0) {
+        Actor_Kill(thisx);
+        return;
+    }
+
+    osSyncPrintf( "女神リフト CT\n"); //Goddess lift CT
+    BgJyaLiftInitHelper(THIS, globalCtx, &D_0600D7E8, 0);
+    Actor_ProcessInitChain(thisx, &D_8089A044);
+    if (Flags_GetSwitch(globalCtx, (thisx->params & 0x3F)) != 0) {
+        BgJyaLiftSetFinalPosY(thisx); //Set final position
+    } else {
+        BgJyaLiftSetInitPosY(thisx);//Set initial position
+    }
+    thisx->room = -1;
+    D_8089A020 = 1;
+    THIS->unk16A = 1;
+}
+
+void BgJyaLift_Destroy(BgJyaLift* this, GlobalContext* globalCtx) {
+    if (this->unk16A != 0) {
+        osSyncPrintf("女神リフト DT\n");//Goddess Lift DT
+        D_8089A020 = 0;
+        DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    }
+}
+
+
+void BgJyaLiftSetInitPosY(BgJyaLift* this) {
+    this->actionFunc = &func_80899D38;
+    
+    this->dyna.actor.posRot.pos.y = 1613.0f;
+    this->unk168 = 0;
+}
+
+
+void func_80899D38(BgJyaLift* this, GlobalContext* globalCtx) {
+    if ((Flags_GetSwitch(globalCtx, (this->dyna.actor.params & 0x3F)) != 0) || (this->unk168 > 0)) {
+
+        this->unk168 = (this->unk168 + 1);
+        if (this->unk168 >= 0x14) {
+
+            func_800800F8(globalCtx, 0xD66, -0x63, &this->dyna.actor, 0);
+            func_80899DBC(this);
+        }
+    } else {
+    }
+}
+
+void func_80899DBC(BgJyaLift* this) {
+    this->actionFunc = &BgJyaLiftMovementHandler;
+}
+
+void BgJyaLiftMovementHandler(BgJyaLift* this, GlobalContext* globalCtx) {
+    f32 distFromBottom;
+    {//Using a scope here made it match.  Not the best solution but it works.
+        f32 tempVelocity;
+        Math_SmoothScaleMaxMinF(&this->dyna.actor.velocity.y, 4.0f, 0.1f, 1.0f, 0.0f); // 0.10000000149011612f
+        tempVelocity = (this->dyna.actor.velocity.y < 0.2f) ? 0.2f : this->dyna.actor.velocity.y;
+        distFromBottom = Math_SmoothScaleMaxMinF(&this->dyna.actor.posRot.pos.y, 973.0f, 0.1f, tempVelocity, 0.2f); // 0.10000000149011612f 0.20000000298023224f
+    }
+    if ((this->dyna.actor.posRot.pos.y < 1440.0f) && (1440.0f <= this->dyna.actor.pos4.y)) {
+            func_8005B1A4(ACTIVE_CAM);
+        // Likely moves the camera down into the main room when the platform moves to y=1440;
+    }
```

Comment:
id say its probably best to leave out comments like this. the function name will give away what its doing when the function is named properly